_posts/2018-03-27-roboporo.md

```diff
@@ -0,0 +1,32 @@
+---
+layout: post
+title: RoboPoro
+date: 2018-03-27
+categories:
+  - projects
+description: A picture serving utility for Discord
+image: https://wallup.net/wp-content/uploads/2016/01/266728-League_of_Legends-Poro-Master_Yi.jpg
+image-sm: https://i.imgur.com/1DH5sC6.png
+author: Kenneth Nero
+author-image: https://i.imgur.com/H63oe1B.png
+author-bio: A first year computing security student at RIT
+author-email: kenero2@gmail.com
+author-social:
+  github: https://github.com/KennethNero
+  linkedin: https://www.linkedin.com/in/knero/
+---
+
+[RoboPoro](https://github.com/KennethNero/RoboPoro) is a [Discord](https://discordapp.com/) utility bot designed (originally) to allow for users to store quick little pictures and recall them at will. As with all things, the scope of this project quickly grew from its intended purpose of single command functionality, into a creation that now supports multiple concurrent shard and server instances each with isolated, dynamically generated storage files.
+
+## The Concept
+The story of this project is one best told in parts, its original concept coming from a dare between myself and another computing security student. It was to be a race to see who could create a simple bot program quicker, and with both of us not being the strongest of coders, I was somewhat certain that I could both take home a win in our little dare, and morph its outcome into a major project. At first my idea was simple *"I'll just make a quick little bot that draws from a hardcoded state and displays a picture linked with it in chat"*, I thought, and as I set out to do just this, I ran into a few problems.
+
+The first problem I ran into was the choice of coding language to use. I'm all one for innovation and breaking out of one's own bounds, however with Java as an option, I figured my final project would be much better refined if I began to write in something I had something of a base in. With the platform chosen, my next hurdle was finding an API to utilize Discord's app interface, and after a rather randomization of choices, I settled on [Discord4J](https://github.com/Discord4J/Discord4J). This was the first time in my experience that I implemented external API through Maven in my Java code, and after a bit of tooling around and research, I managed to accomplish my first goal - the Discord bot that would spit out a picture on command... That wasn't quite impressive enough however.
```

Comment:
"I'm all for innovation and breaking out of one's comfort zone; however, with Java as an option, I figured my final project would be much more refined if I used a language I already had some level of comfort with."

Reply:
"With the platform chosen, my next hurdle was finding a Discord API to use, and choosing blindly from the available options, I settled on [Discord4J](https://github.com/Discord4J/Discord4J).

Reply:
`s/implemented/used an/`
`s/the Discord bot/a Discord bot/`